packages/autocomplete/src/index.ts

```diff
@@ -398,4 +400,51 @@ function filterShellAPI(
   return hits;
 }
 
-export default completer;
+type AutocompleteShellInstanceState = {
+  getAutocompleteParameters: () => AutocompleteParameters;
+  getAutocompletionContext: () => AutocompletionContext;
+};
+
+function transformAutocompleteResults(
+  line: string,
+  results: { result: string }[]
+): [string[], string] {
+  return [results.map((result) => result.result), line];
+}
+
+export type CompletionResults =
+  | [string[], string, 'exclusive']
+  | [string[], string];
+
+export async function initNewAutocompleter(
```

Comment:
I factored this out here because we have to do it for cli-repl and the browser one. The function we return is going to have to merge together the direct command completer results and the mongodb-ts-autocomplete results and I figured it is best to just do that once. Hence this thin wrapper that closes over the lazily imported & initialised state.